core/src/main/scala-3/sttp/tapir/internal/CaseClass.scala

```diff
@@ -86,6 +86,15 @@ class CaseClassField[Q <: Quotes, T](using val q: Q, t: Type[T])(
     case _ => report.throwError(s"Cannot extract annotation: @${annSymbol.name}, from field: ${symbol.name}, of type: ${Type.show[T]}")
   }
 
+  def extractTreeAndOptTreeFromAnnotation(annSymbol: Symbol): Option[(Tree, Tree)] = constructorField.getAnnotation(annSymbol).map {
+    case Apply(_, List(t, TypeApply(Select(_, "$lessinit$greater$default$2"), _))) => (t, '{ None }.asTerm)
```

Comment:
out of curiosity, why not just return the second element of the list, which will correspond to the `encoded` annotation parameter?

Reply:
Currently this method is used in `EndpointAnnotationMacros` where the first parameter is required.

Reply:
I think my question is, why wouldn't this work:

```scala
def extractTreeAndOptTreeFromAnnotation(annSymbol: Symbol): Option[(Tree, Tree)] = constructorField.getAnnotation(annSymbol).map {
  case Apply(_, List(t, u)) => (t, u)
}
```

Reply:
Yes, I've added the method to cover the cases with 1 or 2 parameters. I've described all these cases in [a new test-case](https://github.com/softwaremill/tapir/blob/6b5d195b9305d0715933b40a2d883855eec252d7/core/src/test/scala/sttp/tapir/annotations/DeriveEndpointIOTest.scala#L156).

Actually, only the first parameter is currently used, the second one being ignored. So as an alternative I've been considering a different method - returning first arg and simply ignoring the rest.

I've created [another PR](https://github.com/softwaremill/tapir/pull/1879) with this alternative. The patter matching there is indeed much simpler.

Reply:
> I think my question is, why wouldn't this work: (...)

I wanted to make sure, that the second parameter was Optional. Now I think I might have overcomplicated it :/

Reply:
Happens :)